main.js

```diff
@@ -616,6 +616,12 @@ function _userQuery(msg, callback) {
                 client.execute(msg.message, (err, rows /* , fields */) => {
                     if (rows && rows.rows) rows = rows.rows;
                     returnClientToPool(client);
+                    //convert ts for postgresql and ms sqlserver
+                    if (!err && rows && rows[0].ts && typeof rows[0].ts === 'string') {
+                        for (let i = 0; i < rows.length; i++) {
+                            rows[i].ts = parseFloat(rows[i].ts);
```

Comment:
Is the Really a float? Should be  better parseInt, or?

Reply:
Additionally it seems we need another guard (https://github.com/ioBroker/ioBroker.sql/runs/5838765177?check_suite_focus=true#step:6:910)

... also check for rows[0] before accessing rows[0].ts

Reply:
upps, of course it schould be parseInt(), thank you! 
i was assuming when we have rows we also have at least rows[0] too, don't we?

what is a guard? i can see some errors but honestly don't know what to do with them. However, would appreciate if you could guide me thru with some hints so i can do better in future.

Reply:
i did what you suggested and tests have all passed now (22/22 checks ok). Outstanding changes to Readme.md will follow in a separate PR later/end of week

Reply:
>  i was assuming when we have rows we also have at least rows[0] too, don't we?

Only if there is a response ... empty set also do not contain a [0] element

Guard in this case I mean a check of existenceof rows[0] before accessing a property of it

Reply:
ah.... ok, thanks for explanation